tests/bgp/test_bgp_gr_helper.py

```diff
@@ -0,0 +1,105 @@
+import time
+import pytest
+import logging
+import ipaddress
+from common.helpers.assertions import pytest_assert
+from common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+def test_bgp_gr_helper_routes_perserved(duthost, nbrhosts, setup_bgp_graceful_restart):
+    """
+    Verify that DUT routes are preserved when peer performed graceful restart
+    """
+
+    config_facts  = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    bgp_neighbors = config_facts.get('BGP_NEIGHBOR', {})
+    metadata = config_facts.get('DEVICE_METADATA', {})
+    po = config_facts.get('PORTCHANNEL', {})
+    dev_nbr = config_facts.get('DEVICE_NEIGHBOR', {})
+
+    rtinfo_v4 = duthost.get_ip_route_info(ipaddress.ip_address(u'0.0.0.0'))
+    if len(rtinfo_v4['nexthops']) == 0:
+        pytest.fail("there is no next hop for v4 default route")
```

Comment:
you are mixing if .. fail and pytest_assert. Might as well use pytest_assert here. and a few lines below.